great_expectations/analytics/anonymizer.py

```diff
@@ -0,0 +1,13 @@
+from __future__ import annotations
+
+import secrets
+from hashlib import md5
+
+
+class Anonymizer:
+    def __init__(self, salt: str | None = None) -> None:
+        self._salt = salt or secrets.token_hex(8)
+
+    def anonymize(self, string_: str) -> str:
+        salted = self._salt + string_
+        return md5(salted.encode("utf-8")).hexdigest()
```

Comment:
Ported over whatever logic from the current OSS analytics I thought made sense here